### PR TITLE
Remove some unnecessary dependencies on setuptools, wheel, and poetry

### PR DESCRIPTION
**Pull request by tjni**
## Description of changes

Patch the projects owned by [bachya](https://github.com/bachya) so that they no longer declare unnecessary dependencies on setuptools, wheel, and poetry (they only need poetry-core).

Some of these projects have newer versions released, but I have refrained from updating them to keep things simple: my purpose is only to unbreak these projects after https://github.com/NixOS/nixpkgs/pull/248866 (with the exception of `aiorecollect`, for which my patch submitted did not merge cleanly).

In the same group as https://github.com/NixOS/nixpkgs/pull/250687 and https://github.com/NixOS/nixpkgs/pull/250819 before I realized they are related.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tjni** [comment]
@ofborg build python310Packages.aioguardian python311Packages.aioguardian
@ofborg build python310Packages.aionotion python311Packages.aionotion
@ofborg build python310Packages.aiopurpleair python311Packages.aiopurpleair
@ofborg build python310Packages.aiorecollect python311Packages.aiorecollect
@ofborg build python310Packages.aioridwell python311Packages.aioridwell
@ofborg build python310Packages.aiowatttime python311Packages.aiowatttime
@ofborg build python310Packages.py17track python311Packages.py17track
@ofborg build python310Packages.pyoutbreaksnearme python311Packages.pyoutbreaksnearme
@ofborg build python310Packages.pyiqvia python311Packages.pyiqvia
@ofborg build python310Packages.pyopenuv python311Packages.pyopenuv
@ofborg build python310Packages.pytile python311Packages.pytile
@ofborg build python310Packages.regenmaschine python311Packages.regenmaschine
@ofborg build python310Packages.seventeentrack python311Packages.seventeentrack
@ofborg build python310Packages.simplisafe-python python311Packages.simplisafe-python

File: pkgs/development/python-modules/py17track/default.nix [hunk 1]
```diff
@@ -26,6 +27,26 @@ buildPythonPackage rec {
...
+      url = "https://github.com/bachya/py17track/commit/3b52394759aa50c62e2a56581e30cdb94003e2f1.patch";
+      hash = "sha256-iLgklhEZ61rrdzQoO6rp1HGZcqLsqGNitwIiPNLNHQ4=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace 'attrs = ">=19.3,<21.0"' 'attrs = ">=19.3,<22.0"' \
+      --replace 'async-timeout = "^3.0.1"' 'async-timeout = ">=3.0.1,<5.0.0"'
+  '';
```
Comment (natsukium) [inline pkgs/development/python-modules/py17track/default.nix:48]:
```suggestion
```
These no longer seem to be needed.

**Review by natsukium — APPROVED** [review]
I looked at all the patches, and they were perfect.
LGTM

File: pkgs/development/python-modules/py17track/default.nix (hunk 1 above)
Comment (tjni) [reply pkgs/development/python-modules/py17track/default.nix:48]: Great catch! Let me take a look and see if this can be cleaned up.

Comment (tjni) [reply pkgs/development/python-modules/py17track/default.nix]: It works, and both `py17track` and `seventeentrack` (a fork from it) no longer need `async-timeout` also. I made the changes, and I'll kick off an ofborg build of those two again. Thank you for all your help reviewing.

**Comment by tjni** [comment]
@ofborg build python310Packages.py17track python311Packages.py17track
@ofborg build python310Packages.seventeentrack python311Packages.seventeentrack

**Review by natsukium — APPROVED** [review]

**Pull request merged**